include/cu.hpp

```diff
@@ -250,6 +250,10 @@ class DeviceMemory : public Wrapper<CUdeviceptr> {
     checkCudaCall(cuMemHostGetDevicePointer(&_obj, hostMemory, 0));
   }
 
+  void zero(size_t size);
```

Comment:
```suggestion
  void zero(const size_t size);
```
Would it make sense to do this? Or does that cause trouble?

Reply:
A const before a parameter of some basic type does not do anything useful, does it?  The parameter is passed by value, so from the function user's perspective, the value will not change anyway, with or without a const.

Reply:
Good point, but I think it would protect you from you accidentally changing the parameter in the function though? Or is it the convention not to do this for function arguments.

Reply:
I do not know what the general convention is, but in my opinion, the function prototype is a contract between the user and the implementation of the function.  For call-by-reference parameters, there is an obvious difference between const and nonconst arguments, where the implementation promises not to alter the parameter, but for call-by-value parameters, the const is an implementation detail that the user of the function should not be bothered with.  But this is just my opinion ...